README.md

```diff
@@ -69,17 +69,64 @@ application restarts.
 
 ### Use correct application data directories
 
-Modify Qt application main function by setting the organization and application name
-to match the declaration added to the desktop file.
+Applications that are not based on the Sailfish App library
+([libsailfishapp](https://github.com/sailfishos/libsailfishapp))
+must ensure that correct organization and application name is set.
 
-    QScopedPointer<QGuiApplication> app(Sailfish::createApplication(argc, argv));
     ...
+    QGuiApplication *app = ...;
     app->setOrganizationName(QStringLiteral("org.foobar"));
     app->setApplicationName(QStringLiteral("MyApp"));
     ...
 
-When Sailfish App library (libsailfishapp) is used these values are set automatically to the values
-set in desktop file.
+When Sailfish App library is used, these values are set automatically to the values
+set in application's Desktop Entry file.
+
+Sandboxed applications cannot use `QSettings` constructed with the default arguments - it would use
+an inaccessible path outside `QStandardPaths::AppDataLocation`. A suitable settings file path needs
+to be determined explicitly.
+
+    const QString settingsPath =
+        QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)
+        + "/" + QCoreApplication::applicationName() + ".conf";
+    QSettings settings(settingsPath, QSettings::NativeFormat);
+
+Unless you are creating a new application, it is also necessary to take care about migrating
+application data and configuration from the old location outside the sandbox.
+
+An example of how `QSettings`-based configuration may be migrated:
```

Comment:
Migration on the spec & post install script would be a good example as well.

Reply:
Those work on system level and don't act as the user. It's easier to do it as the running app.

Reply:
Would work for each. We have done it similarly on spec for some Jolla apps.